e2e/suites/provider/cases/common/common.go

```diff
@@ -605,3 +605,48 @@ func DeletionPolicyDelete(f *framework.Framework) (string, func(*framework.TestC
 		}
 	}
 }
+
+func DecodingPolicySync(f *framework.Framework) (string, func(*framework.TestCase)) {
+	return "[common] should sync a secret containing multiple Decoding policies .data[]", func(tc *framework.TestCase) {
+		secretKey1 := fmt.Sprintf("%s-%s", f.Namespace.Name, "one")
+		secretKey2 := fmt.Sprintf("%s-%s", f.Namespace.Name, "other")
+		secretValue1 := "bazz"
+		secretValue2 := "eWFkYQ=="
+		tc.Secrets = map[string]framework.SecretEntry{
+			secretKey1: {Value: secretValue1},
+			secretKey2: {Value: secretValue2},
+		}
+		tc.ExpectedSecret = &v1.Secret{
+			Type: v1.SecretTypeOpaque,
+			Data: map[string][]byte{
+				secretKey1: []byte(secretValue1),
+				secretKey2: []byte("yada"),
+			},
+		}
+		tc.ExternalSecret.Spec.Data = []esv1beta1.ExternalSecretData{
+			{
+				SecretKey: secretKey1,
+				RemoteRef: esv1beta1.ExternalSecretDataRemoteRef{
+					Key:              secretKey1,
+					DecodingStrategy: esv1beta1.ExternalSecretDecodeAuto,
+				},
+			},
+			{
+				SecretKey: secretKey2,
+				RemoteRef: esv1beta1.ExternalSecretDataRemoteRef{
+					Key:              secretKey2,
+					DecodingStrategy: esv1beta1.ExternalSecretDecodeAuto,
```

Comment:
Wouldn't it be better to have another strategy here so the test is a bit more relevant?

Reply:
Done. Testing explicitly now `Auto`, `Base64` and `None`.